### PR TITLE
Rename attributes and improve enhanced views API

### PR DESCRIPTION
**Pull request by jasper-jager**
Attributes were renamed to match the latest specs. The API is improved to add references as embedded HAL JSON in enhanced views

### CODE REVIEW

File: src/gobapi/storage.py [hunk 1]
```diff
@@ -171,11 +171,40 @@ def _get_convert_for_table(table, filter={}):
...
+        return hal_entity
+
+    # Get all metadata or reference fields and filter them from the columns returned by the database view
     metadata_column_list = [k for k in filter.keys()]
-    columns = [c for c in table.columns if c.name not in metadata_column_list]
+    columns = [c for c in table.columns
+               if c.name not in metadata_column_list
+               and not c.name.startswith(('_ref', '_mref'))]
+    reference_columns = [c for c in table.columns if c.name.startswith(('_ref', '_mref'))]
+
```
Comment (Vegter) [inline src/gobapi/storage.py]: I propose to add a method isReference(column) that checks for the name starting with _ref, _mref

File: src/gobapi/storage.py [hunk 2]
```diff
@@ -171,11 +171,40 @@ def _get_convert_for_table(table, filter={}):
...
+    # Create the list of references
+    references = {}
+    for c in reference_columns:
+        # Column name is in the form of '_ref_attribute_name_catalog_collection'
+        column_name_array = c.name.split('_')
+        attribute_name = '_'.join(column_name_array[2:-2])
```
Comment (Vegter) [inline src/gobapi/storage.py]: I propose to add an example that explains this logic

File: src/gobapi/storage.py [hunk 3]
```diff
@@ -171,11 +171,40 @@ def _get_convert_for_table(table, filter={}):
...
+
+        catalog_abbreviation = str(column_name_array[-2])
+        collection_abbreviation = str(column_name_array[-1])
+        ref = GOBModel().get_reference_by_abbreviations(catalog_abbreviation, collection_abbreviation)
+        gob_type = 'GOB.ManyReference' if c.name.startswith('_mref') else 'GOB.Reference'
+
```
Comment (Vegter) [inline src/gobapi/storage.py]: Maybe the _ref and _mref logic should be part of core gobmodel or gob type?

File: src/gobapi/storage.py (hunk 2 above)
Comment (jasper-jager) [reply src/gobapi/storage.py]: Docstring with explanation was added

File: src/gobapi/storage.py (hunk 3 above)
Comment (jasper-jager) [reply src/gobapi/storage.py]: For now I've kept it in GOB-API as it's the only place I think we'll encounter it. If this changes we can always move it

**Pull request merged**